src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/OutputTypeEnumProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework.XamlTypes;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// Provides the mapping between OutputType msbuild property (exposed in the browse object through OutputTypeEx)
+    /// and OutputType property on the BrowseObject.
+    /// Winmdobj and appcontainerexe need to be mapped to dll and exe respectively for the OutputTypeproperty.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("OutputTypeEnumProvider")]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class OutputTypeEnumProvider : IDynamicEnumValuesProvider
+    {
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair> options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(new OutputTypeEnumValuesGenerator());
+        }
+
+        internal class OutputTypeEnumValuesGenerator : IDynamicEnumValuesGenerator
+        {
```

Comment:
What does DisplayName convey here ? Name and Display name doesn't look consistent